seismostats/analysis/bvalue/positive.py

```diff
@@ -55,17 +63,37 @@ def calculate(self,
                                  delta_m=delta_m,
                                  weights=weights)
 
-    def _estimate(self, dmc: float | None = None) -> float:
-
-        # only take the values where the next earthquake is d_mc larger than the
-        # previous one. delta_m is added to avoid numerical errors
+    def _filter_magnitudes(self) -> np.ndarray:
+        '''
+        Filter out magnitudes below the completeness magnitude.
+        '''
+        idx = super()._filter_magnitudes()
+
+        if self.times is not None:
+            self.times = self.times[idx]
+        return idx
```

Comment:
since you created self.idx in filtermagnitudes:
```
super()._filter_magnitudes()
if self.times is not None:
    self.times = self.times[self.idx]
return self.idx
```

this way it is clear what is happening also from just looking at the implementation and not also the base class.